### ISSUE TITLE
Non-bmp (astral) unicode characters confuse the editor

### ISSUE THREAD

**Issue by 7dd6c116-645a-435e-94e2-8d13a1a2de79** [python/cpython#83307]
BPO | [39126](https://bugs.python.org/issue39126)
--- | :---
Nosy | @terryjreedy, @dmaxime
Files | <li>[python3ide bug.mp4](https://bugs.python.org/file48801/python3ide%20bug.mp4 "Uploaded as video/mp4 at 2019-12-23.21:11:54 by @dmaxime")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/terryjreedy'
closed_at = None
created_at = <Date 2019-12-23.21:11:54.374>
labels = ['expert-IDLE', 'type-bug', '3.9']
title = 'Non-bmp (astral) unicode characters confuse the editor'
updated_at = <Date 2019-12-23.22:22:38.532>
user = 'https://github.com/dmaxime'
```

bugs.python.org fields:
```python
activity = <Date 2019-12-23.22:22:38.532>
actor = 'terry.reedy'
assignee = 'terry.reedy'
closed = False
closed_date = None
closer = None
components = ['IDLE']
creation = <Date 2019-12-23.21:11:54.374>
creator = 'dmaxime'
dependencies = []
files = ['48801']
hgrepos = []
issue_num = 39126
keywords = []
message_count = 2.0
messages = ['358836', '358838']
nosy_count = 2.0
nosy_names = ['terry.reedy', 'dmaxime']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue39126'
versions = ['Python 3.9']
```

</p></details>

**Comment by terryjreedy** [comment]
Closed #88383 (with even more comments) as a duplicate of this.

**Comment by serhiy-storchaka** [comment]
This is a limitation of Tcl/Tk 8.6, which by default handles Unicode in 16-bit units, so a non-BMP character occupies two positions in the text widget — a surrogate pair on Windows, a split UTF-8 sequence elsewhere (a Tcl built with `TCL_UTF_MAX=6`, as some Linux distributions do, does not have the problem). Tkinter already converts strings in the best way possible for each platform. IDLE could perhaps work around it, but the code would be very complex and might not always work. Tk 9 uses code points, and with it the example works as expected; the Windows and macOS installers ship Tk 9.0 since Python 3.14. We could keep this open until Tk 9 is used on every platform, but it is very unlikely we will do anything on our side.

**Comment by terryjreedy** [comment]
I agree with closing this.  The problem is solved on Windows and macOS and any linux system that provides tcl/tk 9 at least as an option.
